File: java/src/main/java/com/authdog/resources/ApiSecretsResource.java
package com.authdog.resources;

import com.authdog.AuthdogClient;
import com.authdog.exceptions.ApiException;
import com.authdog.exceptions.AuthenticationException;
import com.fasterxml.jackson.databind.JsonNode;

/**
 * Environment API secret operations.
 */
public final class ApiSecretsResource {
    /**
     * Authdog client.
     */
    private final AuthdogClient client;

    /**
     * Create an API secrets helper.
     * @param clientParam Authdog client
     */
    public ApiSecretsResource(final AuthdogClient clientParam) {
        this.client = clientParam;
    }

    /**
     * List API secrets for an environment.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @return secrets envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode list(final String tenantId,
                         final String environmentId)
            throws AuthenticationException, ApiException {
        return client.request("GET",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/api-secrets",
                null, null, JsonNode.class);
    }

    /**
     * Create an API secret.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param body request body
     * @return created secret envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode create(final String tenantId,
                           final String environmentId,
                           final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/api-secrets",
                body, null, JsonNode.class);
    }

    /**
     * Revoke an API secret.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param secretId secret ID
     * @return revoke envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode revoke(final String tenantId,
                           final String environmentId,
                           final String secretId)
            throws AuthenticationException, ApiException {
        return client.request("POST",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/api-secrets/" + secretId + "/revoke",
                null, null, JsonNode.class);
    }
}

File: csharp/Resources/ApiSecretsResource.cs
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

namespace Authdog
{
    /// <summary>
    /// Environment API secret Wave 2 operations
    /// </summary>
    public class ApiSecretsResource
    {
        private readonly AuthdogClient _client;

        public ApiSecretsResource(AuthdogClient client)
        {
            _client = client;
        }

        public Task<JObject> ListAsync(string tenantId, string environmentId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Get,
                $"{AuthdogClient.Env(tenantId, environmentId)}/api-secrets");

        public JObject List(string tenantId, string environmentId) =>
            ListAsync(tenantId, environmentId).GetAwaiter().GetResult();

        public Task<JObject> CreateAsync(string tenantId, string environmentId, object body) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Post,
                $"{AuthdogClient.Env(tenantId, environmentId)}/api-secrets",
                body);

        public JObject Create(string tenantId, string environmentId, object body) =>
            CreateAsync(tenantId, environmentId, body).GetAwaiter().GetResult();

        public Task<JObject> RevokeAsync(string tenantId, string environmentId, string secretId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Post,
                $"{AuthdogClient.Env(tenantId, environmentId)}/api-secrets/{secretId}/revoke");

        public JObject Revoke(string tenantId, string environmentId, string secretId) =>
            RevokeAsync(tenantId, environmentId, secretId).GetAwaiter().GetResult();
    }
}
